### ISSUE TITLE
[Bug]: Qwen2.5-7b can run but stuck with any reques

### ISSUE THREAD

**Issue by yeungtuzi** [MooreThreads/vllm_musa#7]
### Your current environment

Same environment as https://github.com/MooreThreads/vllm_musa/issues/6
but use Qwen2.5-7B-Instruct instead.

The engine can finally get up after a while, but stuck at first request.

### 🐛 Describe the bug

Run log

> root@ubuntu-s4000-8:/workspace# bash qwen2.5.sh 
> DEBUG 05-16 06:20:58 main.py:48] Setting VLLM_WORKER_MULTIPROC_METHOD to 'spawn'
> INFO 05-16 06:20:58 __init__.py:30] Available plugins for group vllm.platform_plugins:
> INFO 05-16 06:20:58 __init__.py:32] name=musa, value=vllm_musa:register
> INFO 05-16 06:20:58 __init__.py:34] all available plugins for group vllm.platform_plugins will be loaded.
> INFO 05-16 06:20:58 __init__.py:36] set environment variable VLLM_PLUGINS to control which plugins to load.
> INFO 05-16 06:20:58 __init__.py:44] plugin musa loaded.
> INFO 05-16 06:20:58 __init__.py:198] Platform plugin musa is activated
> DEBUG 05-16 06:20:58 __init__.py:28] No plugins for group vllm.general_plugins found.
> INFO 05-16 06:20:58 api_server.py:912] vLLM API server version 0.7.4.dev0+ged6e9075d.d20250418
> INFO 05-16 06:20:58 api_server.py:913] args: Namespace(subparser='serve', model_tag='/home/model/Qwen/Qwen2.5-7B-Instruct', config='', host=None, port=8000, uvicorn_log_level='info', allow_credentials=False, allowed_origins=['*'], allowed_methods=['*'], allowed_headers=['*'], api_key=None, lora_modules=None, prompt_adapters=None, chat_template=None, chat_template_content_format='auto', response_role='assistant', ssl_keyfile=None, ssl_certfile=None, ssl_ca_certs=None, ssl_cert_reqs=0, root_path=None, middleware=[], return_tokens_as_token_ids=False, disable_frontend_multiprocessing=False, enable_request_id_headers=False, enable_auto_tool_choice=False, enable_reasoning=False, reasoning_parser=None, tool_call_parser=None, tool_parser_plugin='', model='/home/model/Qwen/Qwen2.5-7B-Instruct', task='auto', tokenizer=None, skip_tokenizer_init=False, revision=None, code_revision=None, tokenizer_revision=None, tokenizer_mode='auto', trust_remote_code=True, allowed_local_media_path=None, download_dir=None, load_format='auto', config_format=<ConfigFormat.AUTO: 'auto'>, dtype='auto', kv_cache_dtype='auto', max_model_len=16384, guided_decoding_backend='xgrammar', logits_processor_pattern=None, model_impl='auto', distributed_executor_backend=None, pipeline_parallel_size=1, tensor_parallel_size=1, max_parallel_loading_workers=None, ray_workers_use_nsight=False, block_size=None, enable_prefix_caching=None, disable_sliding_window=False, use_v2_block_manager=True, num_lookahead_slots=0, seed=0, swap_space=4, cpu_offload_gb=0, gpu_memory_utilization=0.9, num_gpu_blocks_override=None, max_num_batched_tokens=None, max_num_partial_prefills=1, max_long_partial_prefills=1, long_prefill_token_threshold=0, max_num_seqs=None, max_logprobs=20, disable_log_stats=False, quantization=None, rope_scaling=None, rope_theta=None, hf_overrides=None, enforce_eager=True, max_seq_len_to_capture=8192, disable_custom_all_reduce=False, tokenizer_pool_size=0, tokenizer_pool_type='ray', tokenizer_pool_extra_config=None, limit_mm_per_prompt=None, mm_processor_kwargs=None, disable_mm_preprocessor_cache=False, enable_lora=False, enable_lora_bias=False, max_loras=1, max_lora_rank=16, lora_extra_vocab_size=256, lora_dtype='auto', long_lora_scaling_factors=None, max_cpu_loras=None, fully_sharded_loras=False, enable_prompt_adapter=False, max_prompt_adapters=1, max_prompt_adapter_token=0, device='auto', num_scheduler_steps=1, multi_step_stream_outputs=True, scheduler_delay_factor=0.0, enable_chunked_prefill=None, speculative_model=None, speculative_model_quantization=None, num_speculative_tokens=None, speculative_disable_mqa_scorer=False, speculative_draft_tensor_parallel_size=None, speculative_max_model_len=None, speculative_disable_by_batch_size=None, ngram_prompt_lookup_max=None, ngram_prompt_lookup_min=None, spec_decoding_acceptance_method='rejection_sampler', typical_acceptance_sampler_posterior_threshold=None, typical_acceptance_sampler_posterior_alpha=None, disable_logprobs_during_spec_decoding=None, model_loader_extra_config=None, ignore_patterns=[], preemption_mode=None, served_model_name=['Qwen2.5-7B'], qlora_adapter_name_or_path=None, otlp_traces_endpoint=None, collect_detailed_traces=None, disable_async_output_proc=False, scheduling_policy='fcfs', scheduler_cls='vllm.core.scheduler.Scheduler', override_neuron_config=None, override_pooler_config=None, compilation_config=None, kv_transfer_config=None, worker_cls='auto', generation_config=None, override_generation_config=None, enable_sleep_mode=False, calculate_kv_scales=False, additional_config=None, disable_log_requests=False, max_log_len=None, disable_fastapi_docs=False, enable_prompt_tokens_details=False, dispatch_function=<function ServeSubcommand.cmd at 0x7fbc2f9f4040>)
> DEBUG 05-16 06:20:58 api_server.py:190] Multiprocessing frontend to use ipc:///tmp/06a40c80-3ccc-427b-83c9-3224b8189418 for IPC Path.
> INFO 05-16 06:20:58 api_server.py:209] Started engine process with PID 8228
> INFO 05-16 06:21:04 __init__.py:30] Available plugins for group vllm.platform_plugins:
> INFO 05-16 06:21:04 __init__.py:32] name=musa, value=vllm_musa:register
> INFO 05-16 06:21:04 __init__.py:34] all available plugins for group vllm.platform_plugins will be loaded.
> INFO 05-16 06:21:04 __init__.py:36] set environment variable VLLM_PLUGINS to control which plugins to load.
> INFO 05-16 06:21:04 __init__.py:44] plugin musa loaded.
> INFO 05-16 06:21:04 __init__.py:198] Platform plugin musa is activated
> DEBUG 05-16 06:21:04 __init__.py:28] No plugins for group vllm.general_plugins found.
> INFO 05-16 06:21:05 config.py:549] This model supports multiple tasks: {'classify', 'embed', 'reward', 'score', 'generate'}. Defaulting to 'generate'.
> WARNING 05-16 06:21:05 _custom_ops.py:21] Failed to import from vllm._C with ModuleNotFoundError("No module named 'vllm._C'")
> WARNING 05-16 06:21:06 triton_decode_attention.py:44] The following error message 'operation scheduled before its operands' can be ignored.
> DEBUG 05-16 06:21:06 decorators.py:109] Inferred dynamic dimensions for forward method of <class 'vllm.model_executor.models.deepseek_v2.DeepseekV2Model'>: ['input_ids', 'positions', 'intermediate_tensors', 'inputs_embeds']
> INFO 05-16 06:21:13 config.py:549] This model supports multiple tasks: {'reward', 'classify', 'embed', 'generate', 'score'}. Defaulting to 'generate'.
> WARNING 05-16 06:21:13 _custom_ops.py:21] Failed to import from vllm._C with ModuleNotFoundError("No module named 'vllm._C'")
> WARNING 05-16 06:21:14 triton_decode_attention.py:44] The following error message 'operation scheduled before its operands' can be ignored.
> DEBUG 05-16 06:21:14 decorators.py:109] Inferred dynamic dimensions for forward method of <class 'vllm.model_executor.models.deepseek_v2.DeepseekV2Model'>: ['input_ids', 'positions', 'intermediate_tensors', 'inputs_embeds']
> INFO 05-16 06:21:14 llm_engine.py:234] Initializing a V0 LLM engine (v0.7.4.dev0+ged6e9075d.d20250418) with config: model='/home/model/Qwen/Qwen2.5-7B-Instruct', speculative_config=None, tokenizer='/home/model/Qwen/Qwen2.5-7B-Instruct', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config=None, tokenizer_revision=None, trust_remote_code=True, dtype=torch.bfloat16, max_seq_len=16384, download_dir=None, load_format=auto, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=True, kv_cache_dtype=auto,  device_config=musa, decoding_config=DecodingConfig(guided_decoding_backend='xgrammar'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None, collect_model_forward_time=False, collect_model_execute_time=False), seed=0, served_model_name=Qwen2.5-7B, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=False, chunked_prefill_enabled=False, use_async_output_proc=True, disable_mm_preprocessor_cache=False, mm_processor_kwargs=None, pooler_config=None, compilation_config={"splitting_ops":[],"compile_sizes":[],"cudagraph_capture_sizes":[],"max_capture_size":0}, use_cached_outputs=True, 
> WARNING 05-16 06:21:15 logger.py:202] VLLM_TRACE_FUNCTION is enabled. It will record every function executed by Python. This will slow down the code. It is suggested to be used for debugging hang or crashes only.
> INFO 05-16 06:21:15 logger.py:206] Trace frame log is saved to /tmp/root/vllm/vllm-instance-1d976/VLLM_TRACE_FUNCTION_for_process_8228_thread_140332545482752_at_2025-05-16_06:21:15.084956.log
> DEBUG 05-16 06:21:15 config.py:3461] enabled custom ops: Counter()
> DEBUG 05-16 06:21:15 config.py:3463] disabled custom ops: Counter()
> DEBUG 05-16 06:21:16 parallel_state.py:810] world_size=1 rank=0 local_rank=0 distributed_init_method=tcp://192.168.100.103:59085 backend=mccl
> INFO 05-16 06:21:16 model_runner.py:1110] Starting to load model /home/model/Qwen/Qwen2.5-7B-Instruct...
> DEBUG 05-16 06:21:17 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:21:27 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:21:29 config.py:3461] enabled custom ops: Counter({'rms_norm': 57, 'silu_and_mul': 28, 'rotary_embedding': 1})
> DEBUG 05-16 06:21:29 config.py:3463] disabled custom ops: Counter()
> Loading safetensors checkpoint shards:   0% Completed | 0/4 [00:00<?, ?it/s]
> Loading safetensors checkpoint shards:  25% Completed | 1/4 [00:02<00:06,  2.28s/it]
> Loading safetensors checkpoint shards:  50% Completed | 2/4 [00:04<00:04,  2.07s/it]
> DEBUG 05-16 06:21:35 utils.py:154] Loaded weight lm_head.weight with shape torch.Size([152064, 3584])
> Loading safetensors checkpoint shards:  75% Completed | 3/4 [00:06<00:02,  2.06s/it]
> DEBUG 05-16 06:21:37 client.py:192] Waiting for output from MQLLMEngine.
> Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:08<00:00,  2.02s/it]
> Loading safetensors checkpoint shards: 100% Completed | 4/4 [00:08<00:00,  2.05s/it]
> 
> INFO 05-16 06:21:39 model_runner.py:1115] Loading model weights took 14.2487 GB
> DEBUG 05-16 06:21:47 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:21:57 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:22:07 client.py:192] Waiting for output from MQLLMEngine.
> INFO 05-16 06:22:13 executor_base.py:111] # musa blocks: 32318, # CPU blocks: 4681
> INFO 05-16 06:22:13 executor_base.py:116] Maximum concurrency for 16384 tokens per request: 31.56x
> DEBUG 05-16 06:22:17 client.py:192] Waiting for output from MQLLMEngine.
> INFO 05-16 06:22:17 llm_engine.py:436] init engine (profile, create kv cache, warmup model) took 37.73 seconds
> DEBUG 05-16 06:22:27 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:22:37 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:22:46 engine.py:135] Starting Startup Loop.
> DEBUG 05-16 06:22:46 engine.py:137] Starting Engine Loop.
> DEBUG 05-16 06:22:46 api_server.py:282] vLLM to use /tmp/tmpvhh3z_b7 as PROMETHEUS_MULTIPROC_DIR
> INFO 05-16 06:22:46 api_server.py:958] Starting vLLM API server on http://0.0.0.0:8000
> INFO 05-16 06:22:46 launcher.py:23] Available routes are:
> INFO 05-16 06:22:46 launcher.py:31] Route: /openapi.json, Methods: GET, HEAD
> INFO 05-16 06:22:46 launcher.py:31] Route: /docs, Methods: GET, HEAD
> INFO 05-16 06:22:46 launcher.py:31] Route: /docs/oauth2-redirect, Methods: GET, HEAD
> INFO 05-16 06:22:46 launcher.py:31] Route: /redoc, Methods: GET, HEAD
> INFO 05-16 06:22:46 launcher.py:31] Route: /health, Methods: GET
> INFO 05-16 06:22:46 launcher.py:31] Route: /ping, Methods: POST, GET
> INFO 05-16 06:22:46 launcher.py:31] Route: /tokenize, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /detokenize, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /v1/models, Methods: GET
> INFO 05-16 06:22:46 launcher.py:31] Route: /version, Methods: GET
> INFO 05-16 06:22:46 launcher.py:31] Route: /v1/chat/completions, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /v1/completions, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /v1/embeddings, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /pooling, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /score, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /v1/score, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /v1/audio/transcriptions, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /rerank, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /v1/rerank, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /v2/rerank, Methods: POST
> INFO 05-16 06:22:46 launcher.py:31] Route: /invocations, Methods: POST
> INFO:     Started server process [8157]
> INFO:     Waiting for application startup.
> INFO:     Application startup complete.
> DEBUG 05-16 06:22:47 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:22:56 client.py:171] Heartbeat successful.
> DEBUG 05-16 06:22:56 metrics.py:455] Avg prompt throughput: 0.0 tokens/s, Avg generation throughput: 0.0 tokens/s, Running: 0 reqs, Swapped: 0 reqs, Pending: 0 reqs, GPU KV cache usage: 0.0%, CPU KV cache usage: 0.0%.
> DEBUG 05-16 06:22:56 engine.py:195] Waiting for new requests in engine loop.
> DEBUG 05-16 06:22:57 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:23:06 client.py:171] Heartbeat successful.
> DEBUG 05-16 06:23:06 client.py:171] Heartbeat successful.

Sample request

> curl http://0.0.0.0:8000/v1/chat/completions -H "Content-Type: application/json" -d '{
>     "model": "Qwen2.5-7B",
>     "messages": [
>         {"role": "system", "content": "You are a helpful assistant."},
>         {"role": "user", "content": "1+1等于几？"}
>     ]
> }'

Then the engine stucks

> DEBUG 05-16 06:23:46 engine.py:195] Waiting for new requests in engine loop.
> INFO 05-16 06:23:47 chat_utils.py:332] Detected the chat template content format to be 'string'. You can set `--chat-template-content-format` to override this.
> INFO 05-16 06:23:47 logger.py:39] Received request chatcmpl-a92f5c2ae85a4e05818c92886c8b35d9: prompt: '<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\n1+1等于几？<|im_end|>\n<|im_start|>assistant\n', params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=1.0, top_p=1.0, top_k=-1, min_p=0.0, seed=None, stop=[], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=16359, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None), prompt_token_ids: None, lora_request: None, prompt_adapter_request: None.
> INFO 05-16 06:23:47 engine.py:280] Added request chatcmpl-a92f5c2ae85a4e05818c92886c8b35d9.
> DEBUG 05-16 06:23:47 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:23:56 client.py:171] Heartbeat successful.
> DEBUG 05-16 06:23:57 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:24:06 client.py:171] Heartbeat successful.
> DEBUG 05-16 06:24:07 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:24:16 client.py:171] Heartbeat successful.
> DEBUG 05-16 06:24:17 client.py:192] Waiting for output from MQLLMEngine.
> DEBUG 05-16 06:24:26 client.py:171] Heartbeat successful.
 (and repeat for ever)

**Comment by yeahdongcn** [comment]
This issue has been closed due to inactivity.
If you still encounter this issue or would like to continue the discussion, please provide the requested information or additional details and we will re-open the issue.
